pymbolic/primitives.py

```diff
@@ -181,7 +190,7 @@ def disable_subscript_by_getitem():
     pass
 
 
-class Expression(ABC):
+class Expression:
```

Comment:
`ABC` is not smart enough to understand classes modified by decorators, see `augment_attrs_expression_class` below.

Reply:
Can those two deprecations not be added to the base class directly?

Reply:
No, for compatibility reasons, I think. `Expression` needs to tolerate subclasses that are not yet dataclasses.